src/DNA.cpp

```diff
@@ -489,3 +549,108 @@ string DNA::get_binary_representation(int value, int bit_count) {
 
     return output;
 }//end get_binary_representation
+
+/**
+ * rand_num
+ * 
+ * Returns a random integer [start, end).
+ **/
+int DNA::rand_num(int start, int end) {
+    int diff = end - start;
+    int r = rand() % diff;
+    r += start;
+    return r;
+}//end rand_num
+
+/**
+ * clip
+ * 
+ * Returns only the bits of value between [start, end). 
+ * Assumed that start <= end <= 64;
+ **/
+dna_t DNA::clip(dna_t value, int start, int end) {
+    long s = 0xffffffffffffffff << start;
+    long e;
+    if (end == 64 ) {
```

Comment:
(This is a clarification comment, not a change suggestion). What are you doing here? Does the size of the end value change because of the presence of the signed bit?

Reply:
From what I could tell, there is no _logical_ shift in C++. So, to get around the fact that shifting a negative number leaves 1's at the most significant end, I had to make the most significant bit a zero. Unless, there was no shifting to be done. Does that answer the question?